functions/xnetwork/main.k

```diff
@@ -0,0 +1,77 @@
+import models.io.upbound.azure.network.v1beta1 as networkv1beta1
+import models.io.upbound.azure.network.v1beta2 as networkv1beta2
+import models.io.upbound.azure.v1alpha1 as azurev1alpha1
+import models.io.upbound.azure.v1beta1 as azurev1beta1
+import models.io.upbound.platform.azure.v1alpha1 as platformazurev1alpha1
+import models.k8s.apimachinery.pkg.apis.meta.v1 as metav1
+
+# observed composite resource
+oxr = option("params").oxr
+params = oxr.spec.parameters
```

Comment:
Did you tried to have oxr casted to the model of the XR ? ala:

`oxr = xstoragebucketv1alpha1.XStorageBucket{**option("params").oxr}`

Reply:
@haarchri thanks for the tip, amended 👍